packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/AppTests.kt

```diff
@@ -292,42 +302,41 @@ class AppTests {
 //        assertEquals(registry, app.getFunctions(user, registry).defaultCodecRegistry)
 //    }
 //
-//    @Test
-//    fun encryption() {
-//        val context = InstrumentationRegistry.getInstrumentation().targetContext
-//
-//        // Create new test app with a random encryption key
-//        val testApp = TestApp(appName = TEST_APP_2, builder = {
-//            it.encryptionKey(TestHelper.getRandomKey())
-//        })
-//
-//        try {
-//            // Create Realm in order to create the sync metadata Realm
-//            var user = testApp.login(Credentials.anonymous())
-//
-//            val syncConfig = SyncConfiguration
-//                .Builder(user, "foo")
-//                .testSchema(SyncStringOnly::class.java)
-//                .build()
-//
-//            Realm.getInstance(syncConfig).close()
-//
-//            // Create a configuration pointing to the metadata Realm for that app
-//            val metadataDir = File(context.filesDir, "mongodb-realm/${testApp.configuration.appId}/server-utility/metadata/")
-//            val config = RealmConfiguration.Builder()
-//                .name("sync_metadata.realm")
-//                .directory(metadataDir)
-//                .build()
-//            assertTrue(File(config.path).exists())
-//
-//            // Open the metadata realm file without a valid encryption key
-//            assertFailsWith<RealmFileException> {
-//                DynamicRealm.getInstance(config)
-//            }
-//        } finally {
-//            testApp.close()
-//        }
-//    }
+
+    @Test
+    fun encryption() {
+        // Create new test app with a random encryption key
+        val app = TestApp(
+            appName = TEST_APP_FLEX,
+            builder = {
+                it
+                    .encryptionKey(TestHelper.getRandomKey())
+                    .syncRootDirectory("${appFilesDirectory()}/foo")
+            })
+
+        try {
+            // Create Realm in order to create the sync metadata Realm
+            val user = app.createUserAndLogin()
+            val syncConfig = SyncConfiguration
+                .Builder(user, setOf(ParentPk::class, ChildPk::class))
+                .build()
+            Realm.open(syncConfig).close()
+
+            // Create a configuration pointing to the metadata Realm for that app
+            val config = RealmConfiguration
+                .Builder(setOf())
+                .name(SYNC_METADATA_REALM_NAME)
+                .directory(getMetadataDirectory(app))
```

Comment:
For now, `SYNC_METADATA_REALM_NAME` and `getMetadataDirectory()` are only used by this test. I've seen many hardcoded and repeated values elsewhere, but what are your thoughts on extracting these two?

Reply:
When they are just used in one place, I would keep them inlined here. If we ever need to use them from multiple places, we can look into extracting it then. But it is not a big deal.